cppcache/src/SerializationRegistry.hpp

```diff
@@ -320,8 +334,12 @@ class APACHE_GEODE_EXPORT SerializationRegistry {
 
   inline void serialize(const DataSerializable* obj, DataOutput& output,
                         bool isDelta) const {
-    auto id = obj->getClassId();
-    auto dsCode = getSerializableDataDsCode(id);
+    auto&& type = obj->getType();
+    auto typeIterator = theTypeMap.typeToClassId.find(type);
+    auto id = typeIterator->second;
+
+    typeid(*obj).name();
```

Comment:
What is the purpose of this call. Am I missing some side effect of calling this function and not keeping the results?